pandas/core/dtypes/cast.py

```diff
@@ -901,18 +905,19 @@ def maybe_infer_to_datetimelike(value, convert_dates=False):
     if not len(v):
         return value
 
-    def try_datetime(v):
+    def try_datetime(v, require_iso8601=require_iso8601):
         # safe coerce to datetime64
         try:
-            v = tslib.array_to_datetime(v, errors='raise')
```

Comment:
you dont' need to add this require_iso8601 anywhere here, except for in the actual ``to_datetime()`` call, where it should be True.

Reply:
done.